nexus/db-queries/src/db/datastore/instance.rs

```diff
@@ -1003,6 +1005,156 @@ impl DataStore {
+                        InstanceState::NoVmm,
+                        InstanceState::Failed,
+                        InstanceState::Creating,
+                    ];
+
+                    let instance_state = instance_dsl::instance
+                        .filter(instance_dsl::id.eq(authz_instance.id()))
+                        .filter(instance_dsl::time_deleted.is_null())
+                        .select(instance_dsl::state)
+                        .first_async::<InstanceState>(&conn)
+                        .await;
+
+                    match instance_state {
+                        Ok(state) => {
+                            let state_ok = ok_to_reconfigure_instance_states
+                                .contains(&state);
+
+                            if !state_ok {
+                                return Err(err.bail(Error::conflict(
+                                    "instance must be stopped to update",
+                                )));
+                            }
+                        }
+                        Err(diesel::NotFound) => {
+                            // If the instance simply doesn't exist, we
+                            // shouldn't retry. Bail with a useful error.
+                            return Err(err.bail(Error::not_found_by_id(
+                                ResourceType::Instance,
+                                &authz_instance.id(),
+                            )));
+                        }
+                        Err(e) => {
+                            return Err(e);
+                        }
+                    }
+
+                    if let Some(disk_id) = update.boot_disk_id {
+                        // Ensure the disk is currently attached before updating
+                        // the database.
+                        let expected_state = api::external::DiskState::Attached(
+                            authz_instance.id(),
+                        );
+
+                        let attached_disk: Option<Uuid> = disk_dsl::disk
+                            .filter(disk_dsl::id.eq(disk_id))
+                            .filter(
+                                disk_dsl::attach_instance_id
+                                    .eq(authz_instance.id()),
+                            )
+                            .filter(
+                                disk_dsl::disk_state.eq(expected_state.label()),
+                            )
+                            .select(disk_dsl::id)
+                            .first_async::<Uuid>(&conn)
+                            .await
+                            .optional()?;
+
+                        if attached_disk.is_none() {
+                            return Err(err.bail(Error::conflict(
+                                "boot disk must be attached",
+                            )));
+                        }
+                    }
+
+                    // if and when `Update` can update other fields, set them
```

Comment:
i believe this has been renamed?
```suggestion
                    // if and when `Reconfigure` can update other fields, set them
```